src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/TrackerObjectManager.NativeAot.cs

```diff
@@ -0,0 +1,550 @@
+        }
+
+        public static void PegFromTrackerSource(IntPtr pThis)
+        {
+            Marshal.ThrowExceptionForHR((*(delegate* unmanaged<IntPtr, int>**)pThis)[9](pThis));
+        }
+    }
+
+    // Callback implementation of IFindReferenceTargetsCallback
+    internal static unsafe class FindReferenceTargetsCallback
+    {
+        internal static GCHandle s_currentRootObjectHandle;
+
+        [UnmanagedCallersOnly]
+        private static unsafe int IFindReferenceTargetsCallback_QueryInterface(IntPtr pThis, Guid* guid, IntPtr* ppObject)
+        {
+            if (*guid == IID_IFindReferenceTargetsCallback || *guid == IID_IUnknown)
+            {
+                *ppObject = pThis;
+                return HResults.S_OK;
+            }
+            else
+            {
+                return HResults.COR_E_INVALIDCAST;
+            }
+        }
+
+        [UnmanagedCallersOnly]
+        private static unsafe int IFindReferenceTargetsCallback_FoundTrackerTarget(IntPtr pThis, IntPtr referenceTrackerTarget)
+        {
+            if (referenceTrackerTarget == IntPtr.Zero)
+            {
+                return HResults.E_INVALIDARG;
+            }
+
+            if (TryGetObject(referenceTrackerTarget, out object? foundObject))
+            {
+                // Notify the runtime a reference path was found.
+                return TrackerObjectManager.AddReferencePath(s_currentRootObjectHandle.Target, foundObject) ? HResults.S_OK : HResults.S_FALSE;
+            }
+
+            return HResults.S_OK;
+        }
+
+        private static unsafe IntPtr CreateDefaultIFindReferenceTargetsCallbackVftbl()
+        {
+            IntPtr* vftbl = (IntPtr*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(FindReferenceTargetsCallback), 4 * sizeof(IntPtr));
+            vftbl[0] = (IntPtr)(delegate* unmanaged<IntPtr, Guid*, IntPtr*, int>)&IFindReferenceTargetsCallback_QueryInterface;
+            vftbl[1] = (IntPtr)(delegate* unmanaged<IntPtr, uint>)&ComWrappers.Untracked_AddRef;
+            vftbl[2] = (IntPtr)(delegate* unmanaged<IntPtr, uint>)&ComWrappers.Untracked_Release;
+            vftbl[3] = (IntPtr)(delegate* unmanaged<IntPtr, IntPtr, int>)&IFindReferenceTargetsCallback_FoundTrackerTarget;
+            return (IntPtr)vftbl;
+        }
+
+        internal static unsafe IntPtr CreateFindReferenceTargetsCallback()
+        {
+            IntPtr* wrapperMem = (IntPtr*)NativeMemory.Alloc((nuint)sizeof(IntPtr));
+            wrapperMem[0] = CreateDefaultIFindReferenceTargetsCallbackVftbl();
+            return (IntPtr)wrapperMem;
+        }
+    }
+
+    internal readonly struct ComHolder : IDisposable
+    {
+        private readonly IntPtr _ptr;
+
+        internal readonly IntPtr Ptr => _ptr;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public ComHolder(IntPtr ptr)
+        {
+            _ptr = ptr;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
```suggestion
        public ComHolder(IntPtr ptr)
        {
            _ptr = ptr;
        }

```
Unnecessary. The JIT is going inline these trivial methods without any help.